src/actions/index.js

```diff
@@ -0,0 +1,19 @@
+import axios from 'axios';
+import { projectConfig } from '../config';
+
+const API_KEY = projectConfig.API_KEY;
+const ROOT_URL = projectConfig.ROOT_URL;
```

Comment:
I believe the `API_KEY` will be passed to every call of the ROOT_URL. so instead of adding api key to every action, why not place it in the `ROOT_URL` variable? something like following
```javascript
const ROOT_URL = `${projectConfig.ROOT_URL}?appid=${API_KEY}`
```

Reply:
The OpenWeather API expect a request in the format of:
``` ${ROOT_URL}?q=${city}&APPID=${API_KEY} ```
Concatenating the `API_KEY` with `ROOT_URL` before adding the `city` to it results in a `401 (Unauthorized)` response which according to the **API guide** is because of an invalid `API_KEY` (The API_KEY is correct, the ROOT_URL isn't). We also get a `city` when the action is fired.

Reply:
I'm still confused. I don't think the order of the URL matters here. You can simply add `app_id` before the `q=[city]`